src/main/java/apoc/refactor/GraphRefactoring.java

```diff
@@ -299,16 +299,30 @@ public void categorize(
     }
 
     private Node mergeNodes(Node source, Node target, boolean delete) {
-        copyRelationships(source, copyProperties(source, copyLabels(source, target)), delete);
-        if (delete) source.delete();
+        Map<String, Object> propTemp = new HashMap<>();
+        List<String> keys = new ArrayList<>();
+        for(Map.Entry<String, Object> prop : source.getAllProperties().entrySet()){
+            propTemp.put(prop.getKey(), prop.getValue());
+            keys.add(prop.getKey());
+            source.removeProperty(prop.getKey());
+        }
+        copyRelationships(source, copyLabels(source, target), delete);
+        setProperty(propTemp, keys, target);
         return target;
     }
 
+    private void setProperty(Map<String, Object> propTemp, List<String> keys, Node target) {
```

Comment:
@jexp so we remane it `copyProperties` with this signature `private Node copyProperties(Map<String, Object> properties, Node target)` right?